cmake/CMakeLists.txt

```diff
@@ -156,15 +165,15 @@ endif()
 ######################################################
 # packages with special compiler needs or external libs
 ######################################################
-include_directories(${LAMMPS_SOURCE_DIR})
+target_include_directories(lammps PRIVATE ${LAMMPS_SOURCE_DIR})
 
 if(PKG_USER-ADIOS)
   # The search for ADIOS2 must come before MPI because
   # it includes its own MPI search with the latest FindMPI.cmake
   # script that defines the MPI::MPI_C target
   enable_language(C)
   find_package(ADIOS2 REQUIRED)
-  list(APPEND LAMMPS_LINK_LIBS adios2::adios2)
+  target_link_libraries(lammps PRIVATE adios2::adios2)
```

Comment:
Should this be `PUBLIC` for static libraries?

Reply:
I think CMake handles that automatically.